packages/backend/src/plugins/catalog.ts

```diff
@@ -15,18 +15,23 @@
  */
 
 import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
-import { EntityProvider } from '@backstage/plugin-catalog-node';
 import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
 import { Router } from 'express';
 import { PluginEnvironment } from '../types';
+import { DemoEventBasedEntityProvider } from './DemoEventBasedEntityProvider';
 
 export default async function createPlugin(
   env: PluginEnvironment,
-  providers?: Array<EntityProvider>,
 ): Promise<Router> {
+  const { eventBroker, logger } = env;
```

Comment:
As a general pattern we avoid destructing the `env` in the backend wiring because it makes code easier to copy.

Reply:
cool